### PR TITLE
Replace task observer thread with FutureTask subclass

### PR DESCRIPTION
**Pull request by elihart**
I noticed that the observer thread approach that was being used to check when LottieTasks are done has indeterminate delay for reporting finished tasks, since the thread polls the task. It also adds overhead to create a thread for each task.

Digging into it, it looks like we can simply use FutureTask's `done` function to know when the task is done.

With this change the sample app seems to run perfectly, so functionality seems to be the same.

### CODE REVIEW

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/60cf6928af79eb5be133297fe9843f08699a053c-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/60cf6928af79eb5be133297fe9843f08699a053c-android28)

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/4cb345938a774b783ddbf3ac53bda181ad6fa9ea-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/4cb345938a774b783ddbf3ac53bda181ad6fa9ea-android28)

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/2bf503ad4cebd5c3079a7f2fc2fdae002bf97ed0-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/2bf503ad4cebd5c3079a7f2fc2fdae002bf97ed0-android28)

File: lottie/src/main/java/com/airbnb/lottie/LottieTask.java [hunk 1]
```diff
@@ -154,7 +144,7 @@ private synchronized void notifySuccessListeners(T value) {
     }
   }
 
-  private void notifyFailureListeners(Throwable e) {
+  private synchronized void notifyFailureListeners(Throwable e) {
```
Comment (elihart) [inline lottie/src/main/java/com/airbnb/lottie/LottieTask.java]: The call to `notifySuccessListeners` is synchronized, so it seems like this should be too. Not sure if it was just an oversight?

**Comment by elihart** [comment]
Well, I'm at a loss to explain why the happo report has differences. it seems like a result is getting reused somehow?

even more mysterious, some of the comparisons have the correct composition it seems, but there are differences in the frames. I don't see how that could be caused by this change,

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/93110164cc4fc8b0920abb6b94d2861c4adf82f8-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/93110164cc4fc8b0920abb6b94d2861c4adf82f8-android28)

**Comment by elihart** [comment]
Both those links give 404's

**Comment by gpeal** [comment]
@elihart CI always send a link automatically. Haven't had the time to set up failures well. 
I kicked off another build because I noticed that the same duplicate snapshot thing happened on master so it may not be related to yours but just a bug in my coroutines code which was my first attempt at using them 😛

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/ef6abb78536d1b297002d62e64a7604df0ee9e7b-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/ef6abb78536d1b297002d62e64a7604df0ee9e7b-android28)

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/65da5acde2516b2248c5bc4b0b1fa57dc67ef237-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/65da5acde2516b2248c5bc4b0b1fa57dc67ef237-android28)

**Pull request merged**